### PR TITLE
[TE] yyuen/ember upgrade 2.18

### PR DESCRIPTION
**Pull request by justYves**
### What's new: 
- `ember-cli` upgrade from `2.16` to `2.18`
- set the following rules to "warnings" until all fixed cc @aaronucsd 
  -    "ember/avoid-leaking-state-in-ember-objects": ["warn"],
  -   "ember/no-attrs-in-components": ["warn"]
- Fix tests
  - call `this._super(...arguments)` in all `init()`
  - use Snake Case for route queryParams
- `ember-cli-mirage` upgrade from `0.3` to `0.4`
- Start `ember-cli-mirage`'s server for all tests

### Tests:
- no failing tests
- This needs to be tested extensively on staging cc @apucher

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ export default Component.extend({
    * @method bubbleModalInput
    */
   bubbleModalInput() {
-    this.sendAction('inputAction', this.get('reportAnomalyPayload'));
+    this.get('inputAction')(this.get('reportAnomalyPayload'));
```
Comment (igbopie) [inline thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js]: You can do this.inputActions(...) if you want.

File: thirdeye/thirdeye-frontend/app/pods/components/events-table/component.js [hunk 2]
```diff
@@ -52,57 +52,62 @@ export default Component.extend({
...
     later(() => {
-      this._super(args);
+      this._super(...arguments);
     });
   },
 
-  /**
-   * Columns required by ember-models-table
-   */
-  columns: [
```
Comment (igbopie) [inline thirdeye/thirdeye-frontend/app/pods/components/events-table/component.js]: You can use Object.freeze(...)

Comment (justYves) [reply thirdeye/thirdeye-frontend/app/pods/components/events-table/component.js]: oh thanks! didn't know about this work around

File: thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js (hunk 1 above)
Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js]: I personally would keep it `get` as that's more proper way of getting and setting props for Ember.Object.

File: thirdeye/thirdeye-frontend/app/pods/components/anomaly-id/component.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ export default Component.extend({
    * @method anomalyChangeRate
    * @return {Number} - total % change from baseline
    */
-  anomalyChangeRate: computed('anomaly.current', 'anomaly.baseline', function() {
+  anomalyChangeRate: computed('anomaly.{current,baseline}', function() {
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/anomaly-id/component.js]: This is interesting to combined props of the same parent object.

Comment (justYves) [reply thirdeye/thirdeye-frontend/app/pods/components/anomaly-id/component.js]: This is actually the new syntax, ember https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/use-brace-expansion.md

File: thirdeye/thirdeye-frontend/app/pods/components/events-table/component.js [hunk 4]
```diff
@@ -52,57 +52,62 @@ export default Component.extend({
...
-      sortedBy: 'duration',
-      disableFiltering: true
-    }
+  init() {
+    this._super(...arguments);
+
+    /**
+     * Columns required by ember-models-table
+     */
+    const columns = [
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/events-table/component.js]: This init is growing too large. Can we move this outside of init? Like at the top? Or better yet, some const file?

File: thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js (hunk 1 above)
Comment (igbopie) [reply thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js]: That is changing ;)
https://github.com/emberjs/rfcs/blob/master/text/0281-es5-getters.md

Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js]: Man. Ember keeps changing like this. Soon we just do regular vanilla. :)

**Pull request merged**